### ISSUE TITLE
How to know if the ryml::Tree is empty?

### ISSUE THREAD

**Issue by VolodymyrKuksa** [biojppm/rapidyaml#272]
Hello!

I've got this use case for the rapidyaml library:
```
char data[] = "";
ryml::Tree tree = ryml::parse_in_place(ryml::to_substr(data));
std::cout << "tree.empty() = " << tree.empty() << '\n';
```
The output for which is `tree.empty() = 0`. Apparently the `m_size` member of `ryml::Tree` is set to 1 after parsing.
How do I check if the input text contains no yaml nodes? Am I misusing rapidyaml?

The version I'm using is 0.4.1 (the latest release), built as a single header library.

**Comment by biojppm** [comment]
Yes, this is a nagging corner case that requires attention. IIRC the part will always create the root node this carrying this behavior.

As a workaround, you can query the number of children of the root node.

**Comment by VolodymyrKuksa** [comment]
Got it, thanks for the quick reply!
Awesome library!

**Comment by biojppm** [comment]
Let's not close this yet. This issue is something that should be addressed, at the very least by clarifying it in the docs, and preferably by having a behavior that is less surprising.
